### DISCUSSION TITLE
SAE lässt sich nicht hinter Reverse Proxy betreiben

### DISCUSSION THREAD

**Discussion by allcoolusernamesaregone** [camueller/SmartApplianceEnabler#563] [Installation und Betrieb]
Hallo,

ich würde den SAE gerne hinter einem Reverse Proxy (Nginx) betreiben, 

a) für leichten Zugriff von außen
b) mit entspr. IP und Passwortschutz durch Nginx,

entsprechend erreichbar unter  https://meinedomain.de/sae/    (sae aus dieser URL soll das spätere X sein...)

Leider funktioniert das nicht, mindestens durch diese Stellen, die auf "Root" verweisen.
- Startseite: <base hrfe="/">
- vendor.991983e8d4113489.js:   constructor(http, prefix = "/assets/i18n/", suffix = ".json") {
-  sae-service.ts:  public static API = SaeService.BASE_URL + '/sae';

Nach dem "Patchen" der ersten beiden Stellen läuft es weiter, bis zum absoluten Aufruf von /sae/appliances, statt wenn es relativ wäre /sae/sae/appliances..   (dito info, settings, ...). Wie man diese TS Datei im WAR ändern könnte, k.A. ;)

Wäre es möglich, entweder

a) diese (und ggf. noch mehr?) Stellen auf relative Pfade/URLs umzubauen (oder muss das wirklich absolut sein)?
b) falls absolut nötig, bzw. Umbau mit erheblichen folgen, Einführen eines Parameters X, welcher ergänzt wird, z.B.

- Startseite: <base hrfe=X+"/">
- vendor.991983e8d4113489.js:   constructor(http, prefix = X+"/assets/i18n/", suffix = ".json") {
-  sae-service.ts:  public static API = SaeService.BASE_URL + 'X'+'/sae';

mit b) wäre weiterhin alles absolut, aber müsste das Problem lösen können....

**Comment by camueller** [comment]
Von meiner Seite werde ich keinen Aufwand in das Thema stecken, aber wenn Du einen PR bereitstellst, werde ich mit diesen gerne ansehen.

**Comment by allcoolusernamesaregone** [comment]
Es läuft übrigens, wenn man statt einer "Sub-URL" /sae   einfach eine Subdomain sae.meinedomain.de nutzt.
Über nginx z.B.:

map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      close;
}

server {
    listen 443      ssl http2;
    server_name     sae.domain.com;
    include         snippets/certbot.conf;
    ssl_protocols   SSLv3 TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers     HIGH:!aNULL:!MD5;    
    proxy_buffering off;
    location / {
      proxy_pass http://192.168.x.y:8181/;
      proxy_redirect http:// https://;
      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Host $server_name;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection $connection_upgrade;
    }
}

Ob und wie nginx authentifiziert/autorisiert ist dann nochmal was anderes...
